src/Form/DataTransformer/ImageTransformer.php

```diff
@@ -25,65 +24,35 @@ public function __construct(RequestStack $requestStack, EntityManagerInterface $
         $this->requestStack = $requestStack;
         $this->entityManager = $entityManager;
     }
-    // TODO: może o tym trzeba decydować w twigu po prostu??
     public function transform($imageAddress)
     {
-        // dd("transform method used");
-        // dd($imageAddress);
-        // $qb = $this->entityManager->createQueryBuilder();
-        // dd($qb
-        //     ->select('a')
-        //     ->from('App\Entity\Article', 'a')
-        //     ->where($qb->expr()->eq('a.id', $this->requestStack->getCurrentRequest()->get('id')))
-        //     ->getQuery()
-        //     ->getSingleResult()
-        //     ->getImage());
-        if ($imageAddress === null) {# or ""?
+        if ($imageAddress === null) {
             dump('transform - imageAddress = null');
             return null;
-            // return new File('uploads/images/default_image.jpeg');
         }
         dump('transform - imageAddress != null');
         return new File($imageAddress);
     }
 
     public function reverseTransform($imageFile)
     {
-        // dd($imageFile);
         $articleRepository = $this->entityManager->getRepository(Article::class);
         $articleId = $this->requestStack->getCurrentRequest()->get('id');
         if ($imageFile === null) {#or ""..?
             dump('reveerseTransform - imageFile = null');
             return ($articleId === null) ? null : $articleRepository->find($articleId)->getImage();
-            // return $qb #TODO: tutaj entityManager->getRepository(Article::class)->find($this->requestStack->getCurrentRequest()->get('id'))->getImage();
-            //     ->select('a')
-            //     ->from('App\Entity\Article', 'a')
-            //     ->where($qb->expr()->eq('a.id', $this->requestStack->getCurrentRequest()->get('id')))
-            //     ->getQuery()
-            //     ->getSingleResult()
-            //     ->getImage();
-
-            // dd($qb
-            //     ->select('a')
-            //     ->from('App\Entity\Article', 'a')
-            //     ->where($qb->expr()->eq('a.id', $this->requestStack->getCurrentRequest()->get('id')))
-            //     ->getQuery()
-            //     ->getSingleResult()
-            //     ->getImage());
         }
         dump('reverseTransform - imageFile != null');
         if ($articleId !== null) {
             $imageAddress = $articleRepository->find($articleId)->getImage();
-            if ($imageAddress !== null) {
+            if ($imageAddress !== null && $imageAddress !== 'uploads/images/default_image.jpeg') {
                 $fileSystem = new Filesystem();
-                $fileSystem->remove($imageAdress);
+                $fileSystem->remove($imageAddress);
             }
         }
         $directory = 'uploads/images/';
         $imageName = md5(uniqid()).'.'.$imageFile->guessExtension();
         $imageFile->move($directory, $imageName);
```

Comment:
Do obsługi upload'u powinna być utworzona osobna usługa. Teraz gdybyś w innym miejsce w aplikacji potrzebował również obsłużyć upload zdjęć musiałbyś zduplikować ten kod. Nowo utworzony serwis powinien robić całą robotę i wtedy miałbyś np. `uploaderHelper->upload($file)`, gdzie metoda `UploaderHelper::upload()` zwraca stringa (lokalizację obiektu).

Po drugie transformer to nie jest miejsce do zapisywania plików. Teraz mógłbym 100x wysłać nieprawidłowy formularz i utworzyło by mi na dysku 100 zdjęć. Obsługa plików w formularzach najczęściej polega na dodaniu pola typu `FileType` lub `ImageType` z opcją `mapped => false` i obsłużeniem tego osobno.

Proponuję, abyś nie próbował tego refaktoryzować tylko od nowa napisać ze świeżym podejściem (wcześniej poszukaj na internecie jakiś porad).